WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift

```diff
@@ -64,34 +64,40 @@ struct ProductFormActionsFactory: ProductFormActionsFactoryProtocol {
     private let variationsPrice: VariationsPrice
 
     private let stores: StoresManager
+    private let featureFlagService: FeatureFlagService
 
     private let isLinkedProductsPromoEnabled: Bool
     private let linkedProductsPromoCampaign = LinkedProductsPromoCampaign()
     private var linkedProductsPromoViewModel: FeatureAnnouncementCardViewModel {
         .init(analytics: ServiceLocator.analytics,
               configuration: linkedProductsPromoCampaign.configuration)
     }
-    private let isCustomFieldsEnabled: Bool
+
+    private var isCustomFieldsEnabled: Bool {
+        /// There's a technical API limitation where product ID is required to save custom fields,
+        /// thus it needs to be disabled during new product creation
+        featureFlagService.isFeatureFlagEnabled(.viewEditCustomFieldsInProductsAndOrders)
+        && formType != .add
```

Comment:
❓ There are 3 form types: add, edit, and readonly. Do we want to display custom fields for readonly mode? If so, should we add an option to disable editing in this mode?

Reply:
We decided to not show custom fields in readonly mode, and this is clarified further in https://github.com/woocommerce/woocommerce-ios/pull/14068/commits/722f9782fc62663e7b03052b55265f72df358862